### PR TITLE
[10.0][ADD]  Exception type as warning

### PR DESCRIPTION
**Pull request by TDu**
Allow to tag an exception as a warning. Contrary to a regular exception
a warning will be raised but not be blocking.

### CODE REVIEW

File: base_exception/models/base_exception.py [hunk 1]
```diff
@@ -62,6 +63,16 @@ class ExceptionRule(models.Model):
 #  - uid: current user id
 #  - context: current context
 """)
+    warning_only = fields.Boolean(
+        string='Warning only',
```
Comment (Tonow-c2c) [inline base_exception/models/base_exception.py:57]: Translation already added ?

**Review by grindtildeath — COMMENTED** [review]
LGTM :+1: 
Can you please rebase and increase module's version number ?

**Comment by bealdav** [comment]
ping @hparfr

**Review by hparfr — COMMENTED** [review]
Hi,

I don't see why you need this.
You can already check a subset of rules in your own module with `_rule_domain`.
Blocking or not if there is some exceptions can be done in your own `test_exceptions`.

**Comment by TDu** [comment]
Hello @hparfr, indeed this maybe achievable with a cusom `test_exceptions` method. But it is much more user friendly to be able to flag/see which exception are only warnings or not.
There is also an additional text for the warning which is handy.
One of the use case for this (which is live on a production instance) is on sale order lines. Warning only would be displayed on the line as additional information to not forget but they are not blocking to confirm the sale order.

**Comment by bealdav** [comment]
@TDu thanks for this contribution. On my side I think it's a feature that can be implemented in another module (base_exception <> extended_exception ;-) ). 
Then, you can explain more precisely the use case and how implement it in sale for example.
Thanks a lot

File: base_exception/models/base_exception.py [hunk 2]
```diff
@@ -73,6 +74,16 @@ class ExceptionRule(models.Model):
...
+        default=False
+    )
+    warning_text = fields.Char(
+        string='Warning message',
+        translate=True,
+        help='This message can be added to the record that raised a warning.'
+             'For a sale order line it could be the product name sold.'
```
Comment (yvaucher) [inline base_exception/models/base_exception.py:64]: Missing a space before `For`

**Review by grindtildeath — APPROVED** [review]

**Review by i-vyshnevska — APPROVED** [review]

**Review by hparfr — CHANGES_REQUESTED** [review]
Please create another module in order to keep base_exception module away from the warning logic.
Most of the dependencies (purchase, sale, product?) of base_exception don't have to deal with warnings.

**Pull request closed without merging**